node/impl/full/eth.go

```diff
@@ -1262,6 +1263,59 @@ func (e *EthEvent) EthGetFilterLogs(ctx context.Context, id ethtypes.EthFilterID
 	return nil, xerrors.Errorf("wrong filter type")
 }
 
+func parseBlockRange(heaviest abi.ChainEpoch, fromBlock, toBlock *string, maxRange abi.ChainEpoch) (minHeight abi.ChainEpoch, maxHeight abi.ChainEpoch, err error) {
+	if fromBlock == nil || *fromBlock == "latest" {
+		minHeight = heaviest
+	} else if *fromBlock == "earliest" {
+		minHeight = 0
+	} else {
+		if !strings.HasPrefix(*fromBlock, "0x") {
+			return 0, 0, xerrors.Errorf("FromBlock is not a hex")
+		}
```

Comment:
This seems unnecessarily onerous to me. Maybe we could be more lax on this and just check the strings against a regex or something similar (`/(0x)?[0-9a-f/i`); then if `EthUint64FromHex` requires the `0x` we could prepend it. Or just make `EthUint64FromHex` more relaxed in the process.

Alternatively, at least we should list that it must be prefixed with `0x` in the docs.

Reply:
I changed it so that atleast for the Actor Events API, we prepend this if it does not exist. Don't want to touch the ETH events API here.

Reply:
Got rid of all this by changing the `FromEpoch` and `ToEpoch` types in the filter to `abi.ChainEpoch`.

Reply:
Addressed this by not making "0x" mandatory in the input. The docs do mention that these must be hex encoded.